corehq/apps/app_manager/static/app_manager/js/releases.js

```diff
@@ -7,13 +7,49 @@ function SavedApp(o, r) {
     if (!self.include_media) {
         self.include_media = ko.observable(false);
     }
+    if(!self.generating_url){
+        self.generating_url = ko.observable(false);
+    }
+
+    self.generate_short_url = function(url_type){
+        //accepted_url_types = ['short_odk_url', 'short_odk_media_url', 'short_url']
+        if (!url_type){ url_type = 'short_odk_url'; }
+        var base_url = '/a/' + self.domain() + '/apps/odk/' + self.id() + '/',
+            should_generate_url = ((url_type === 'short_odk_url') && self.short_odk_url && !self.short_odk_url()) ||
```

Comment:
i don't have context on this, but it seems like there's a lot of checking in this statement. is there a way to reduce that? do we conditionally define `self.short_odk_url`?